site/src/pages/WorkspacePage/WorkspaceScheduleControls.test.tsx

```diff
@@ -0,0 +1,189 @@
+import { render, screen } from "@testing-library/react";
+import { ThemeProvider } from "contexts/ThemeProvider";
+import { QueryClient, QueryClientProvider, useQuery } from "react-query";
+import { MockWorkspace } from "testHelpers/entities";
+import { WorkspaceScheduleControls } from "./WorkspaceScheduleControls";
+import { workspaceByOwnerAndName } from "api/queries/workspaces";
+import { RouterProvider, createMemoryRouter } from "react-router-dom";
+import userEvent from "@testing-library/user-event";
+import { server } from "testHelpers/server";
+import { rest } from "msw";
+import dayjs from "dayjs";
+import * as API from "api/api";
+import { GlobalSnackbar } from "components/GlobalSnackbar/GlobalSnackbar";
+
+const Wrapper = () => {
+  const { data: workspace } = useQuery(
+    workspaceByOwnerAndName(MockWorkspace.owner_name, MockWorkspace.name),
+  );
+
+  if (!workspace) {
+    return null;
+  }
+
+  return <WorkspaceScheduleControls workspace={workspace} canUpdateSchedule />;
+};
+
+test("add 3 hours to deadline", async () => {
+  const user = userEvent.setup();
+  const baseDeadline = dayjs().add(3, "hour");
+  const updateDeadlineSpy = jest
+    .spyOn(API, "putWorkspaceExtension")
+    .mockResolvedValue();
+  server.use(
+    rest.get(
+      "/api/v2/users/:username/workspace/:workspaceName",
+      (req, res, ctx) => {
+        return res(
+          ctx.status(200),
+          ctx.json({
+            ...MockWorkspace,
+            latest_build: {
+              ...MockWorkspace.latest_build,
+              deadline: baseDeadline.toISOString(),
+            },
+          }),
+        );
+      },
+    ),
+  );
+  render(
+    <ThemeProvider>
+      <QueryClientProvider client={new QueryClient()}>
+        <RouterProvider
+          router={createMemoryRouter([{ path: "/", element: <Wrapper /> }])}
+        />
+      </QueryClientProvider>
+      <GlobalSnackbar />
+    </ThemeProvider>,
+  );
+  await screen.findByTestId("schedule-controls");
+
+  // Check if base deadline is displayed correctly
+  expect(screen.getByText("Stop in 3 hours")).toBeInTheDocument();
+
+  const addButton = screen.getByLabelText("Add hour");
+  await user.click(addButton);
+  await user.click(addButton);
+  await user.click(addButton);
+  await screen.findByText(
+    "Workspace shutdown time has been successfully updated.",
+  );
+  expect(updateDeadlineSpy).toHaveBeenCalledTimes(1);
```

Comment:
Is this part checking to see if requests are being "debounced"/"aggregated"? GitHub just happened to show me the test file, so right now, I'm a little unclear on how clicking the button three times leads to the mocked function only getting called once

I'm wondering if this behavior could be split off into a separate test case. It feels like it would consolidate some of the logic across some of the tests, while increasing the documentation about the expected behavior

Reply:
> Is this part checking to see if requests are being "debounced"/"aggregated"? 

Yes, exactly. It is a good point, I'm going to work on that.